File: backend/app/models/knowledge_base.py
from sqlalchemy import Column, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.core.database import Base
import uuid


class KnowledgeBase(Base):
    __tablename__ = "knowledge_bases"
    
    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    name = Column(String(100), nullable=False)
    description = Column(String(500))
    collection_name = Column(String(100), nullable=False, unique=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    
    # Relationships
    documents = relationship("Document", back_populates="knowledge_base", cascade="all, delete-orphan")
    conversations = relationship("Conversation", back_populates="knowledge_base")

File: frontend/src/services/knowledgeBase.ts
import api from './api'

export interface KnowledgeBase {
  id: string
  user_id: string
  name: string
  description: string
  collection_name: string
  created_at: string
  updated_at: string
}

export interface CreateKnowledgeBaseData {
  name: string
  description?: string
}

export const knowledgeBaseService = {
  async list(): Promise<KnowledgeBase[]> {
    const response = await api.get<KnowledgeBase[]>('/api/knowledge-bases')
    return response.data
  },

  async get(id: string): Promise<KnowledgeBase> {
    const response = await api.get<KnowledgeBase>(`/api/knowledge-bases/${id}`)
    return response.data
  },

  async create(data: CreateKnowledgeBaseData): Promise<KnowledgeBase> {
    const response = await api.post<KnowledgeBase>('/api/knowledge-bases', data)
    return response.data
  },

  async update(id: string, data: Partial<CreateKnowledgeBaseData>): Promise<KnowledgeBase> {
    const response = await api.put<KnowledgeBase>(`/api/knowledge-bases/${id}`, data)
    return response.data
  },

  async delete(id: string): Promise<void> {
    await api.delete(`/api/knowledge-bases/${id}`)
  },
}